plugin/k8s_external/setup.go

```diff
@@ -29,6 +34,15 @@ func setup(c *caddy.Controller) error {
 			e.externalServicesFunc = x.ExternalServices
 			e.externalSerialFunc = x.ExternalSerial
 		}
+
+		if e.externalFunc == nil ||
+			e.externalAddrFunc == nil ||
+			e.externalSerialFunc == nil ||
+			e.externalServicesFunc == nil {
+			return nil
```

Comment:
I think we could log a warning/error here... 

I don't know if it's appropriate to panic and shutdown though at this point.  I don't think it is.

Reply:
... actually this code is never reached unless there is a plugin named `kubernetes` that does not implement `Externaler`... which basically would never happen in any reasonable scenario.

Reply:
A warning could be logged in the `m == nil` check above.

Reply:
> I don't know if it's appropriate to panic and shutdown though at this point. I don't think it is.

I think we should end the process promptly to avoid users being unaware of their misconfiguration. If the Kubernetes plugin is not configured or Externalizer is not implemented, we can initiate a failure by returning an error.